appinventor/components/src/com/google/appinventor/components/runtime/Web.java

```diff
@@ -1180,4 +1207,166 @@ private CapturedProperties capturePropertyValues(String functionName) {
     }
     return null;
   }
+
+  @Override
+  public YailList getDataValue(YailList key) {
+    // If the last GET task is not yet done/cancelled, then the lastTask.get()
+    // method is invoked to wait for completion of the task.
+    if (lastTask != null && !lastTask.isDone() && !lastTask.isCancelled()) {
+      try {
+        lastTask.get();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      } catch (ExecutionException e) {
+        e.printStackTrace();
+      }
+    }
+
+    // Update the locally stored columns list with the contents of the
+    // last response and response type. The reason why this is not done
+    // after HTTP methods is to not reduce performance for non-Chart related
+    // work done.
+    updateColumns(lastResponse, lastResponseType);
```

Comment:
Column updating can be optimized between getDataValue and notify observers methods